src/local.rs

```diff
@@ -105,29 +92,45 @@ impl Local {
         )
         .context(OpenDatabaseSnafu {})?;
 
-        // Get the relative directory of the maildir to the database path.
-        let canonical_db_path = db.path().canonicalize().context(CanonicalizeSnafu {})?;
-        let canonical_mail_dir_path = mail_dir
-            .as_ref()
+        // Get notmuch's idea of the mail root. If, for whatever reason, we get nothing back for
+        // that key (ancient version of notmuch?), use the database path.
+        let mail_root = db
+            .config(ConfigKey::MailRoot)
+            .map_or(
+                db.path().into(),
+                |root| PathBuf::from(root)
+            )
             .canonicalize()
             .context(CanonicalizeSnafu {})?;
-        let relative_mail_dir = canonical_mail_dir_path
-            .strip_prefix(&canonical_db_path)
-            .context(MailDirNotASubdirOfNotmuchRootSnafu {
-                mail_dir: &canonical_mail_dir_path,
-                notmuch_root: &canonical_db_path,
-            })?;
 
-        // Build the query to search for all mail in our maildir.
-        let all_mail_query = format!("path:\"{}/**\"", relative_mail_dir.to_str().unwrap());
+        // Figure out our maildir. Either the configured thing, or notmuch's mail root. Which in
+        // the worst case will be notmuch's database dir, but that's probably not the worst choice.
+        let mail_dir = match &config.mail_dir {
+            Some(ref dir) => dir.clone(),
+            _             => mail_root.clone(),
+        }.canonicalize().context(CanonicalizeSnafu {})?;
+        debug!("mail dir: {}", mail_dir.to_string_lossy());
+
+        // Build the query to search for all mail in our maildir. If the maildir is under the
+        // notmuch mail root, then search under the relative maildir path (allowing multiple
+        // maildirs per notmuch dir). If not, assume this is the only maildir for the notmuch dir,
+        // and use a global query.
+        let all_mail_query = mail_dir
+            .strip_prefix(&mail_root)
+            .ok()
+            .filter(|rel| rel.components().count() > 0)
+            .map_or(
+                "path:**".to_string(),
```

Comment:
This can probably just be an empty string tbh

Reply:
Needs a bit of rework elsewhere to do so because we append `and lastmod:...` in `Local::all_emails_since` and with empty-string it produces an invalid query.

(probably, that rework is for the allmail query to be an `Option`)